### PR TITLE
Added Divine Intervention

### PR DESCRIPTION
**Pull request by ProfessorBiddle**
This is by far the best and most complicated effect I've made yet.

This effect spawns jesus's at a medium rate far above the player, where they parachute down (mostly, some of them decide to just fall) and begin attacking him with random weapons, and any vehicle they can find.

They are very easy to kill, but can have a wide array of deadly weapons, and will always attack the player.

### CODE REVIEW

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		{
+			Ped& ped = jesuss[i];
+			if (!ped)
+			{
+				ped = jesus;
+				jesusDespawnTime[i] = 5;
+				break;
+			}
+		}	
+		SET_MODEL_AS_NO_LONGER_NEEDED(maxJesus);
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:42]: This line should be removed, as it doesn’t work with that parameter ;)

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		//Give em' a shove
+		APPLY_FORCE_TO_ENTITY_CENTER_OF_MASS(jesus, 0, 35.f, 0, -2500.f, true, false, true, true);
+		//give parachute, for some reason only works with both
+		SET_PED_GADGET(jesus, parachute, true);
+		GIVE_WEAPON_TO_PED(jesus, parachute, 9999, true, true);
+		//Jesus hates criminals
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:48]: „Criminals“ == Everyone?

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, playerGroup);
+		SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, civGroup);
+		SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, femCivGroup);
+		SET_PED_RELATIONSHIP_GROUP_HASH(jesus, relationshipGroup);
+		//random weapons
+		Hash wep = weps[g_random.GetRandomInt(0, weps.size() - 1)];
+		GIVE_WEAPON_TO_PED(jesus, wep, 9999, true, false);
+		SET_CURRENT_PED_WEAPON(jesus, wep, true);
+		//Make jesus badass, but not as badass as Griefer Jesus		
+		SET_ENTITY_PROOFS(jesus, false, true, true, true, false, false, false, false);
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:60]: Why make them explosion-proof? Isn’t that a little too much? :thinking:

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		//random weapons
+		Hash wep = weps[g_random.GetRandomInt(0, weps.size() - 1)];
+		GIVE_WEAPON_TO_PED(jesus, wep, 9999, true, false);
+		SET_CURRENT_PED_WEAPON(jesus, wep, true);
+		//Make jesus badass, but not as badass as Griefer Jesus		
+		SET_ENTITY_PROOFS(jesus, false, true, true, true, false, false, false, false);		
+		SET_PED_CAN_RAGDOLL_FROM_PLAYER_IMPACT(jesus, true);
+		SET_RAGDOLL_BLOCKING_FLAGS(jesus, 5);
+		SET_PED_SUFFERS_CRITICAL_HITS(jesus, true);
+		//Safely parachute to player (sometimes)
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:64]: Shouldn’t they all land safely?

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+		TASK_PARACHUTE_TO_TARGET(jesus, playerPos.x, playerPos.y, playerPos.z);
+	}
+	static DWORD64 lastTick2 = 0;
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:67]: Please rename to be more descriptive

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+					continue;
+				}
+			}
+			jesusAmount--;
+
+			SET_OBJECT_AS_NO_LONGER_NEEDED(&jesus);
+			jesus = 0;
+		}
+	}
+	if (lastTick2 < curTick - 1000)
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:104]: Define a constant for the timespans to be easier to maintain

**Review by Last0xygen — CHANGES_REQUESTED** [review]

File: ChaosMod/Effects/db/Peds/PedsJesusRain.cpp (hunk 4 above)
Comment (ProfessorBiddle) [reply ChaosMod/Effects/db/Peds/PedsJesusRain.cpp:64]: I can't figure out how to make them all land safely, they should but some of them always remove their parachutes and fall

**Comment by ProfessorBiddle** [comment]
Holy shit my last commit broke everything. Rewriting effect from the ground up, a better version will be PR'd soon.

**Pull request closed without merging**